jobs/pre-stop-script/spec

```diff
@@ -0,0 +1,13 @@
+name: pre-stop-script
+
+templates:
+  pre-stop.sh: bin/pre-stop
+
+packages: []
+
+properties:
+  script:
+    description: Script that should run before the VM stops, run as root user.
+    example: |-
+        #!/bin/bash
+        apt-get update && apt-get install wget git tmux -y
```

Comment:
This is a copy/paste [from the `pre-start-script` job](https://github.com/cloudfoundry/os-conf-release/blob/master/jobs/pre-start-script/spec#L13).

This example is not relevant, as installing new software before stopping a Daemon seems awkward.

Please provide some relevant example use-case here.

Reply:
The specific use case i need this for is to delete a google cloud route to a vm, before deleting the same vm (while going through the deploy lifecycle). If I don't do that, the bosh CPI might try delete the vm, by going through itself and subsequently bosh won't receive the return call from google and won't know what the outcome of the operation was.

Reply:
To break it down further:
- we already use a pre-start-script to set up a custom NAT vm instance
- when bosh tries to redeploy, under some conditions, it will try to recreate the vm
- when it does, it makes a delete-vm call via the CPI to the google cloud api
- if the delete call goes through the vm, that we want to delete, the connection is cut off before receiving a response
- we have 2 more vms of the same type in different azs
- there is no way to control which NAT vm bosh will use when making the delete call
- by deleting the routing rule in the say pre-stop lifecycle step, we guarantee that if the vm gets recreated, the delete call will go through one of the other 2 vms.

Reply:
So i hope You understand that giving such specific example in the spec is not the best thing.